### DISCUSSION TITLE
Add multiple media to a collection

### DISCUSSION THREAD

**Discussion by chinmaypurav** [spatie/laravel-medialibrary#3295] [Q&A]
I have a file input that accepts `n` files. I am using `multiple` attribute to accept that `n` file input.

```html
<input type="file" name="avatars" multiple> 
```

I get an array of `UploadedFile` instances.

If it's a single file I can do

```php
$model->addMediaFromRequest('avatar')->toMediaCollection();
```

But how do I do it dynamically for `n` files?

I need an efficient way to handle this situation.

If there is not any way, I am willing to open a PR for the same.

**Comment by chinmaypurav** [comment]
This is a solution I am using for now

```php
private function getDottedKeysArray(array $array, string $baseKey): array
{
    return array_keys(Arr::dot($array, "{$baseKey}."));
}

private function addMutipleFilesFromRequest(Model $model, array $keys, string $collection = 'default'): void
{
    foreach ($keys as $key) {
        $model->addMediaFromRequest($key)
            ->toMediaCollection($collection);
    }
}
```

**Comment by rabrowne85** [comment]
I too was struggling with this just recently and the solution I settled on was as below. 

```php
$model->addMultipleMediaFromRequest(['avatars'])
    ->each(function (FileAdder $fileAdder) {
        $fileAdder->toMediaCollection();
    });
```

This was inspired by the [test](https://github.com/spatie/laravel-medialibrary/blob/fd90eceec5eec3c39bd5334fcaf630c0b84a487d/tests/Feature/FileAdder/IntegrationTest.php#L162) suite for the package.
